components/script/dom/bindings/str.rs

```diff
@@ -113,6 +113,74 @@ pub fn is_token(s: &[u8]) -> bool {
     })
 }
 
+/// Returns whether the language is matched, as defined by
+/// [RFC 4647](https://tools.ietf.org/html/rfc4647#section-3.3.2).
+pub fn extended_filtering(tag: &str, range: &str) -> bool {
+    let lang_ranges: Vec<&str> = range.split(',').collect();
+    let mut is_match: bool = true;
+
+    for lang_range in lang_ranges {
```

Comment:
Instead of using a `for` loop here, you can use [`any`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any), since the overall match succeeds if at least one of the range matches with the tag. You can also eliminate the `is_match` variable this way.